staging/src/k8s.io/client-go/discovery/cached/memcache.go

```diff
@@ -117,35 +119,34 @@ func (d *memCacheClient) Fresh() bool {
 func (d *memCacheClient) Invalidate() {
```

Comment:
godoc is now wrong. did we sweep callers that were built around the currently documented behavior?

Reply:
I looked for NewMemCacheClient calls in the kubernetes/kubernetes repo and all calls are using this as CachedDiscoveryInterface interface which defines Invalidate() as:
```
	// Invalidate enforces that no cached data is used in the future that is older than the current time.
	Invalidate()
```

so we should be good. Godoc needs update.

Reply:
@mborsz, can you add a release note warning about this behavior change, as there might be users outside of kubernetes/kubernetes.

I agree this change doesn't break kuberentes/kubernetes. I followed all the use cases of NewMemCacheClient. Most cases the memCacheClient is used to create DeferredDiscoveryRestmapper, which is compatible with this new Invlidate() semantics. In other cases, the Invalidate() method isn't called.

Reply:
I think that even outside of kubernetes/kubernetes we should be good: NewMemCacheClient returns type discovery.CachedDiscoveryInterface (not memCacheClient), so any correct code using CachedDiscoveryInterface shouldn't assume any specific implementation of that interface and CachedDiscoveryInterface promises that we invalidate data only (without any promise about when the refresh happens).

I agree that adding this to release note is a good idea.

Reply:
Done.